extensions/amp-consent/0.1/consent-ui.js

```diff
@@ -235,21 +284,47 @@ export class ConsentUI {
     toggle(dev().assertElement(this.placeholder_), false);
     toggle(dev().assertElement(this.ui_), true);
     classList.remove('loading');
-    classList.add('i-amphtml-consent-ui-in');
-    classList.add('consent-iframe-active');
+    this.vsync_.mutate(() => {
+      classList.add(consentUiClasses.uiIn);
+      classList.add(consentUiClasses.iframeActive);
+    });
   }
 
   /**
    * Remove the iframe from doc
    * Remove event listener
    * Reset UI state
+   * Takes in a function to call after our transition has ended
+   * @param {Function} callback
    */
-  resetIframe_() {
+  resetIframe_(callback) {
     const {classList} = this.parent_;
-    classList.remove('i-amphtml-consent-ui-in');
+
+    // Remove the iframe active to go back to our 30px height
     classList.remove('consent-iframe-active');
-    this.win_.removeEventListener('message', this.boundHandleIframeMessages_);
-    removeElement(dev().assertElement(this.ui_));
+
+    if (this.isFullscreen_) {
+      this.win_.removeEventListener('message', this.boundHandleIframeMessages_);
+
+      classList.add(consentUiClasses.fullscreenExit);
+      classList.remove(consentUiClasses.fullscreen);
+
+      this.isFullscreen_ = false;
+
+      this.getTransformTransitionEndPromise_().then(() => {
+        classList.remove(consentUiClasses.fullscreenExit);
+        classList.remove(consentUiClasses.uiIn);
+        removeElement(dev().assertElement(this.ui_));
+        callback();
+      });
+    } else {
+      this.win_.removeEventListener('message', this.boundHandleIframeMessages_);
+      classList.remove(consentUiClasses.uiIn);
+      this.getTransformTransitionEndPromise_().then(() => {
```

Comment:
Didn't fully get this, I don't think we need transition when to hide iframe.

Reply:
Let's add a comment here to get a UX review